lint/action.yaml

```diff
@@ -25,27 +22,7 @@ runs:
       if: inputs.checkout-repo == 'true'
       with:
         fetch-depth: 0
-    - name: Setup tools
-      uses: open-turo/action-setup-tools@v1
-    # TODO: Configure .pypi credentials using pypi-token
-    - name: Install poetry
-      if: hashFiles('poetry.lock') != ''
-      shell: bash
-      run: ${GITHUB_ACTION_PATH}/../lib/poetry-install.sh
-    - name: Make sure poetry uses the current python version
-      if: hashFiles('poetry.lock') != ''
-      shell: bash
-      run: |
-        # Use the version that pyenv gives as python version pyenv version-name
-        if ! command -v pyenv &>/dev/null; then
-          echo "pyenv not found, skipping"
-          exit 0
-        fi
-        PYTHON_VERSION_SET=$(pyenv version-name)
-        poetry env use $PYTHON_VERSION_SET
-    - name: Install pip dependencies
```

Comment:
Not sure I understand this part

Reply:
You are not installing pip dependencies on lint. If for whatever reason there is a pre-commit hook that requires dependencies to be installed (any static analysis of the code / linter) then this would fail. I would keep them here and instead see how we could use action caches to reuse pip dependencies installs between jobs

Reply:
pre-commit will load the packages it needs so there should not be any need to install dependencies before calling pre-commit. But we can definitely add the install dependency step. We can also add a cache for poetry https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies which will make things faster as well.